niworkflows/interfaces/cifti.py

```diff
@@ -364,7 +364,7 @@ def _create_cifti_image(
                 annot = nb.load(annotation_files[hemi == "RIGHT"])
                 medial = np.nonzero(annot.darrays[0].data)[0]
             # extract values across volumes
-            ts = np.array([tsarr.data[medial] for tsarr in surf.darrays])
+            ts = np.array([tsarr.data[medial] for tsarr in surf_ts.darrays])
```

Comment:
Oh, here would be a good place to use `agg_data()`.

```suggestion
            ts = surf_ts.agg_data()[medial, :]
```

Reply:
doesn't `agg_data()` return a tuple?

Reply:
If the image has multiple types of data array, or you request multiple intents, then no.

If you want to be safe (e.g., if we start putting surfaces inside functional files...):

```suggestion
            ts = surf_ts.agg_data("time series")[medial, :]
```

Reply:
FWIW this case is explicitly covered in the docs: https://nipy.org/nibabel/reference/nibabel.gifti.html#nibabel.gifti.gifti.GiftiImage.agg_data

Reply:
running fmriprep with this change leads to:
```
   ts = surf_ts.agg_data("time series")[medial, :]
TypeError: tuple indices must be integers or slices, not tuple
```

I think keeping it as it was is simplest ATM

Reply:
Sure. That seems really weird, though. If you can `pdb` it at some point, I would be interested in the details.

Reply:
Oh, if there are no darrays, you do get an empty tuple.